File: python/questions/minimumDeletions.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        de = 0
        for c in s:
            de += ord('b') - ord(c)
        ans = de
        for c in s:
            de += (ord(c)-ord('a'))*2 - 1;
            ans = min(ans, de)
        return ans

File: cpp/questions/minimumDeletions.cpp
#include <vector>
#include <climits>

using namesapce std;

class Solution {
public:
    int minimumDeletions(string s) {
        int del = 0;
        for (auto c: s) {
            del += 'b' - c;
        }
        int ans = del;
        for (auto c: s) {
            del += (c-'a') * 2 - 1;
            ans = std::min(ans, del);
        }
        return ans;
    }
};
